File: spec/functional_test/fixtures/lua/cli_cliargs/tool.lua
local cli = require("cliargs")

cli:set_name("mytool")
cli:add_argument("INPUT", "path to input file")
cli:add_option("-o, --output=DEST", "path to output file", "out.txt")
cli:add_flag("-v, --verbose", "enable verbose output")

local token = os.getenv("API_TOKEN")

local args, err = cli:parse()
print(token, args, err)

File: spec/functional_test/fixtures/zig/cli_yazap/tool.zig
const std = @import("std");
const yazap = @import("yazap");
const App = yazap.App;
const Arg = yazap.Arg;

pub fn main() !void {
    var gpa = std.heap.GeneralPurposeAllocator(.{}){};
    const allocator = gpa.allocator();

    var app = App.init(allocator, "tool", "Example CLI tool");
    defer app.deinit();

    var root_cmd = app.rootCommand();
    try root_cmd.addArg(Arg.booleanOption("verbose", 'v', "Enable verbose output"));
    try root_cmd.addArg(Arg.positional("input", "Input file", null));

    var build_cmd = app.createCommand("build", "Build the project");
    try build_cmd.addArg(Arg.booleanOption("release", 'r', "Build in release mode"));
    try build_cmd.addArg(Arg.singleValueOption("output", 'o', "Output path"));
    try root_cmd.addSubcommand(build_cmd);

    const matches = try app.parseProcess();
    _ = matches;

    const token = std.process.getEnvVarOwned(allocator, "API_TOKEN");
    _ = token;
}
